src/components/CustomCollection/EditableCollectionTitle.tsx

```diff
@@ -1,91 +1,76 @@
-import React, { useState, useRef, useEffect } from 'react'
-import { Textarea } from '@trussworks/react-uswds'
+import React, { useRef, useEffect } from 'react'
+
+import {
+  Button,
+  ButtonGroup,
+  Form,
+  Label,
+  TextInput,
+} from '@trussworks/react-uswds'
 
 import styles from './CustomCollection.module.scss'
-/* eslint-disable jsx-a11y/no-noninteractive-element-to-interactive-role */
+
 type PropTypes = {
   collectionId: string
   text: string
-  onSave: (s: string) => void
+  onSave: (event: React.FormEvent<HTMLFormElement>) => void
+  onCancel: () => void
   onDelete: () => void
+  isEditing: boolean
 }
+
 export const EditableCollectionTitle = ({
   collectionId,
   text,
+  isEditing,
   onSave,
-  onDelete,
+  onCancel,
 }: PropTypes) => {
-  const [isEditing, setEditing] = useState(text === '')
-  const [currentText, setCurrentText] = useState(text)
-  const inputRef = useRef<HTMLTextAreaElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
 
+  const handleCancel = () => {
+    const inputEl = inputRef.current as HTMLInputElement
+    inputEl.value = ''
+    onCancel()
+  }
   useEffect(() => {
     if (isEditing) {
       inputRef?.current?.focus()
     }
   }, [isEditing])
 
-  const handleKeyDown = (event: React.KeyboardEvent) => {
-    const { key } = event
-    const keys = ['Escape', 'Enter', 'Tab']
-
-    if (keys.includes(key)) {
-      if (isEditing) {
-        inputRef?.current?.blur()
-      } else if (key === 'Enter') {
-        setEditing(true)
-      }
-    }
-  }
-
-  const handleOnBlur = () => {
-    if (currentText.length) {
-      onSave(currentText)
-
-      setEditing(false)
-    } else if (text.length) {
-      // Revert to previous value
-      setCurrentText(text)
-      setEditing(false)
-    } else {
-      // No value, delete the collection
-      onDelete()
-    }
-  }
-
   const inputId = `collectionTitle_${collectionId}`
 
   return (
-    <>
+    <div className={styles.editCollectionTitle}>
       {isEditing ? (
-        <>
-          <label htmlFor={inputId} className="usa-sr-only">
+        <Form onSubmit={onSave} className={styles.editableCollectionTitle}>
+          <Label htmlFor={inputId} className="usa-sr-only">
             Collection Title
-          </label>
-          <Textarea
+          </Label>
+          <TextInput
             inputRef={inputRef}
             id={inputId}
-            name="title"
+            name="collectionTitle"
+            required
             maxLength={200}
             className={styles.collectionTitle}
-            value={currentText}
-            placeholder="Add a title for this collection"
-            onKeyDown={(e) => handleKeyDown(e)}
-            onChange={(e) => setCurrentText(e.target.value)}
-            onBlur={handleOnBlur}
+            placeholder="Name this collection"
+            type="text"
```

Comment:
I would expect this input to have a default value of whatever the existing title is (and to reset to that value if the user cancels out). I had to implement a similar form in the Edit Custom Link modal if you want an example: https://github.com/USSF-ORBIT/ussf-portal-client/pull/487/files#diff-7557230bf066bb54052bf830361b2efbeacf03c1e953f31e360c71e8e80ec7aa